crates/hyperqueue/src/client/commands/job.rs

```diff
@@ -174,13 +188,59 @@
     let mut responses =
         rpc_call!(connection, message, ToClientMessage::JobDetailResponse(r) => r).await?;
 
-    if let Some(job) = responses.pop().and_then(|v| v.1) {
-        return gsettings.printer().print_job_output(
-            job,
-            task_selector,
-            output_stream,
-            task_header,
-        );
+    if let Some(mut job) = responses.pop().and_then(|v| v.1) {
+        let all_ids: Vec<JobTaskId> = job.tasks.iter().map(|task| task.task_id).collect();
```

Comment:
Please change this to `Set`, otherwise it would be O(n^2) when checking the task ID, which could be nasty if there are a lot of tasks.